### PR TITLE
Update generator to null safety

### PR DESCRIPTION
**Pull request by Schwusch**
It would be nice to have working tests :)

### CODE REVIEW

File: generator/lib/model/shape.dart [hunk 1]
```diff
@@ -150,10 +150,10 @@ class Shape {
     if (type.isBasicType()) {
       return false;
     }
-    if (member != null && !member.shapeClass.requiresJson) {
+    if (!(member?.shapeClass?.requiresJson == true)) {
```
Comment (isoos) [inline generator/lib/model/shape.dart:153]: Maybe a simpler form?
`if (member?.shapeClass?.requiresJson != true) {`
(same below)

**Review by isoos — APPROVED** [review]
lgtm
there may be some opportunity to help the compiler make earlier decisions about non-nullability, but this looks good, and the micro-optimizations can be done if/when we are touching that part of the code.

File: generator/lib/model/shape.dart (hunk 1 above)
Comment (Schwusch) [reply generator/lib/model/shape.dart]: good eye :+1:

**Comment by Schwusch** [comment]
> there may be some opportunity to help the compiler make earlier decisions about non-nullability, 

Yeah, this was a least effort migration, and I tried to make the amount of changes small.
There's a little more exclamation marks than I'm usually comfortable with.

**Comment by Schwusch** [comment]
This reminds me that we don't have CI tests checking formatting. I'll add that as well.

**Pull request merged**